cypress/integration/core-opensearch-dashboards/opensearch-dashboards/workspace-plugin/mds_workspace_collaborators.spec.js

```diff
@@ -0,0 +1,210 @@
+        permissions: {
+          library_write: {
+            users: [`${Cypress.env('username')}`],
+            groups: ['admin_group', 'new_admin_group'],
+          },
+          write: {
+            users: [`${Cypress.env('username')}`],
+            groups: ['admin_group', 'new_admin_group'],
+          },
+          library_read: {
+            users: ['read_user', 'new_read_user'],
+          },
+          read: {
+            users: ['read_user', 'new_read_user'],
+          },
+        },
+      };
+      cy.checkWorkspace(workspaceId, expectedWorkspace);
+    });
+
+    it('should change access level successfully', () => {
+      cy.get('table')
+        .contains('td', 'read_user')
+        .parent('tr')
+        .within(() => {
+          cy.get('input[type="checkbox"]').check();
+        });
+      cy.get('table')
+        .contains('td', 'admin_group')
+        .parent('tr')
+        .within(() => {
+          cy.get('input[type="checkbox"]').check();
+        });
+
+      cy.getElementByTestId(
+        'workspace-detail-collaborator-table-actions'
+      ).click();
+      cy.get('div[role="dialog"]')
+        .find('span')
+        .contains('Change access level')
+        .click();
+      cy.get('div[role="dialog"]')
+        .find('span')
+        .contains('Read and write')
+        .click();
+      cy.getElementByTestId('confirmModalConfirmButton').click();
+
+      cy.wait(2000); // Intentional Wait
+
+      cy.get('table')
+        .contains('td', 'read_user')
+        .parent()
+        .within(() => {
+          cy.get('td').eq(3).contains('Read and write');
+        });
+      cy.get('table')
+        .contains('td', 'admin_group')
+        .parent()
+        .within(() => {
+          cy.get('td').eq(3).contains('Read and write');
+        });
+      const expectedWorkspace = {
+        name: workspaceName,
+        features: ['use-case-observability'],
+        description: 'test_description',
+        permissions: {
+          library_write: {
+            users: [`${Cypress.env('username')}`, 'read_user'],
+            groups: ['admin_group'],
+          },
+          write: {
+            users: [`${Cypress.env('username')}`],
+          },
+          read: {
+            groups: ['admin_group'],
+            users: ['read_user'],
+          },
+        },
+      };
+      cy.checkWorkspace(workspaceId, expectedWorkspace);
+    });
+
+    it('should delete collaborators successfully', () => {
+      cy.get('table')
+        .contains('td', 'read_user')
+        .parent('tr')
+        .within(() => {
+          cy.get('input[type="checkbox"]').check();
+        });
+      cy.get('table')
+        .contains('td', 'admin_group')
+        .parent('tr')
+        .within(() => {
+          cy.get('input[type="checkbox"]').check();
+        });
+
+      cy.getElementByTestId('confirm-delete-button').click();
+      cy.getElementByTestId('confirmModalConfirmButton').click();
+      cy.wait(2000); // Intentional Wait
```

Comment:
Nit: Shall we need to wait `confirmModalConfirmButton` disappear or check collaborator not exist in the table? It would be more safe to ensure collaborators been deleted.